arangod/RocksDBEngine/RocksDBRecoveryManager.cpp

```diff
@@ -307,10 +306,25 @@ class WBReader final : public rocksdb::WriteBatch::Handler {
       if (hash != 0) {
         uint64_t objectId = RocksDBKey::objectId(key);
         auto est = findEstimator(objectId);
-        if (est.first != nullptr && est.second < currentSeqNum) {
+        if (est != nullptr && est->commitSeq() < currentSeqNum) {
+          /*LOG_TOPIC(TRACE, Logger::ENGINES)
+            << "recovery adding hash'" << hash
+            << "' to estimator for index with objectId'" << objectId << "'";*/
+
           // We track estimates for this index
-          est.first->insert(hash);
-        }
+          est->insert(hash);
+        } /*else {
+          if (est.first == nullptr) {
+            LOG_TOPIC(TRACE, Logger::ENGINES)
+              << "not tracking estimator for index with objectId' "
+              << objectId << "', estimator not found";
+          } else {
+            LOG_TOPIC(TRACE, Logger::ENGINES)
+              << "not tracking estimator for index with objectId' "
+              << objectId << "' (synced at " << est.second << " >= "
+              << currentSeqNum << ")";
+          }
```

Comment:
Can we activate the trace output here?
Should help for debugging if index estimates are off.